packages/cli/src/databases/migrations/common/1724951148974-AddApiKeysTable.ts

```diff
@@ -0,0 +1,55 @@
+import type { ApiKey } from '@/databases/entities/api-key';
+import type { MigrationContext } from '@/databases/types';
+import { generateNanoId } from '@/databases/utils/generators';
+
+export class AddApiKeysTable1724951148974 {
+	async up({
+		queryRunner,
+		escape,
+		runQuery,
+		schemaBuilder: { createTable, column },
+	}: MigrationContext) {
+		const userTable = escape.tableName('user');
+		const userApiKeysTable = escape.tableName('user_api_keys');
+		const userIdColumn = escape.columnName('userId');
+		const apiKeyColumn = escape.columnName('apiKey');
+		const labelColumn = escape.columnName('label');
+		const idColumn = escape.columnName('id');
+
+		// Create the new table
+		await createTable('user_api_keys')
+			.withColumns(
+				column('id').varchar(36).primary,
+				column('userId').uuid.notNull,
+				column('label').varchar(100).notNull,
+				column('apiKey').varchar().notNull,
+			)
+			.withForeignKey('userId', {
+				tableName: 'user',
+				columnName: 'id',
+				onDelete: 'CASCADE',
+			})
+			.withIndexOn(['userId', 'label'], true)
+			.withIndexOn(['apiKey'], true).withTimestamps;
+
+		const usersWithApiKeys = (await queryRunner.query(
+			`SELECT ${idColumn}, ${apiKeyColumn} FROM ${userTable} WHERE ${apiKeyColumn} IS NOT NULL`,
+		)) as Array<Partial<ApiKey>>;
+
+		// Move the apiKey from the users table to the new table
+		usersWithApiKeys.forEach(async (user: { id: string; apiKey: string }) => {
```

Comment:
This does not work as expected. Creating promises inside the `.forEach` causes them to be "detached" from the stack (i.e. the code continues after `forEach` without waiting for the promises to resolve). Correct approach would be to use `await Promise.all(usersWithApiKeys.map(...))`